File: app-data/src/main/java/com/zxjdev/smile/data/user/entity/UserMapper.java
package com.zxjdev.smile.data.user.entity;

import com.zxjdev.smile.domain.common.base.BaseMapper;
import com.zxjdev.smile.domain.user.User;

import javax.inject.Inject;

public class UserMapper extends BaseMapper<User, UserEntity> {

  @Inject
  public UserMapper() {

  }

  @Override
  public User transform(UserEntity userEntity) {
    User user = new User();
    user.setUsername(userEntity.getUsername());
    user.setNickname(userEntity.getNickName());
    user.setAvatar(userEntity.getAvatar());
    return user;
  }
}

File: app-data/src/main/java/com/zxjdev/smile/data/user/entity/UserMapper.kt
package com.zxjdev.smile.data.user.entity

import com.zxjdev.smile.domain.common.base.BaseMapper
import com.zxjdev.smile.domain.user.User

import javax.inject.Inject

class UserMapper @Inject constructor() : BaseMapper<User, UserEntity>() {

    override fun transform(userEntity: UserEntity): User {
        val user = User()
        user.username = userEntity.username
        user.nickname = userEntity.nickName
        user.avatar = userEntity.avatar
        return user
    }
}